examples/p2p-data/script.js

```diff
@@ -0,0 +1,90 @@
+const Peer = window.Peer;
+
+(async function main() {
+  const localId = document.querySelector('[data-local-id]');
+  const localText = document.querySelector('[data-local-text]');
+  const connectTrigger = document.querySelector('[data-connect-trigger]');
+  const closeTrigger = document.querySelector('[data-close-trigger]');
+  const sendTrigger = document.querySelector('[data-send-trigger]');
+  const remoteId = document.querySelector('[data-remote-id]');
+  const messages = document.querySelector('[data-messages]');
+
+  const peer = new Peer({
+    key: window.__SKYWAY_KEY__,
+    debug: 3,
+  });
+
+  // Register connecter handler
+  connectTrigger.addEventListener('click', () => {
+    // Note that you need to ensure the peer has connected to signaling server
+    // before using methods of peer instance.
+    if (!peer.open) {
+      return;
+    }
+
+    const dataConnection = peer.connect(remoteId.value);
+
+    dataConnection.once('open', async () => {
+      messages.textContent += `=== DataConnection has been opened ===\n`;
+
+      sendTrigger.addEventListener('click', onClickSend);
+    });
+
+    dataConnection.on('data', data => {
+      messages.textContent += `Remote: ${data}\n`;
+    });
+
+    dataConnection.once('close', () => {
+      messages.textContent += `=== DataConnection has been closed ===\n`;
+      sendTrigger.removeEventListener('click', onClickSend);
+    });
+
+    // Register closing handler
+    closeTrigger.addEventListener('click', () => dataConnection.close(), {
+      once: true,
+    });
+
+    function onClickSend() {
+      const data = localText.value;
+      dataConnection.send(data);
+
+      messages.textContent += `You: ${data}\n`;
+      localText.value = '';
+    }
+  });
+
+  peer.once('open', id => (localId.textContent = id));
+
+  // Register connectee handler
```

Comment:
nits: To my best knowledge, there's no word 'connectee'. This word may be "connected party".

Reply:
It was just convenient conversion inspired by "caller / callee" relations.
But I fix it later!

Reply:
d45e503